### ISSUE TITLE
Question to <Mutiple-threading -> Condition Variables section -> demo code) 

### ISSUE THREAD

**Issue by Kangaxx-0** [sunface/rust-course#792]
[thread topic](https://course.rs/advance/concurrency-with-threads/thread.html), there is a demo code as below

<img width="740" alt="image" src="https://user-images.githubusercontent.com/85712372/163472143-867e9bbc-0bd0-4ede-aca0-6c96260ca719.png">

The question is why **ampersand**, is it there intentionally?

**Comment by Rustln** [comment]
In Rust language, the following code is equivalent.

```rust
let x = &1;
```
and

```rust
let ref x = 1;
```

-----

```rust
let y = *x;
```

and

```rust
let &y = x;
```

------

The `let &(ref lock, ref cvar) = &*pair2;` code is equivalent to 

```rust
let tuple: &(Mutex<bool>,Condvar) = &*pair2;   // note: type declaration
let (lock, cvar) = *tuple;   // note: *
let lock = &lock;
let cvar = &cvar;
```

This seems a little verbose, so we use the `&` and `ref` syntax to quickly dereference and add references.

**Comment by Kangaxx-0** [comment]
> @Kangaxx-0 In Rust language, the following code is equivalent.
> 
> ```rust
> let x = &1;
> ```
> 
> and
> 
> ```rust
> let ref x = 1;
> ```
> 
> ```rust
> let y = *x;
> ```
> 
> and
> 
> ```rust
> let &y = x;
> ```
> 
> The `let &(ref lock, ref cvar) = &*pair2;` code is equivalent to
> 
> ```rust
> let tuple: &(Mutex<bool>, Condvar) = &*pair2; // note: type declaration
> let (lock, cvar) = *tuple; // note: *
> let lock = &lock;
> let cvar = &cvar;
> ```
> 
> This seems a little verbose, so we use the `&` and `ref` syntax to quickly dereference and add references.

Thanks for this explanation, I understood that line which doing a quick dereference, but I am trying to understand the necessity of references which is the ampersand, in other words,  if I do `let (ref lock, ref cvar) = *pair2;`, it still works, Pair2 get moved into the new thread, then it will be automatically dropped when it is out of the scope, anything I misunderstood?

**Comment by Rustln** [comment]
Yes, in this case, `&` it doesn't seem necessary. Now I can't understand the author's motivation. Your question is right. Of course, you don't have to pay too much attention to this point. Continue to study. I believe you will think clearly in the future. I also look forward to your answer at that time.
